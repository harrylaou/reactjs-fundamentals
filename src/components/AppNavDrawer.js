import React, {Component} from 'react'
import Drawer from 'material-ui/Drawer'
import MenuItem from 'material-ui/MenuItem'
import { withRouter } from 'react-router'

class AppNavDrawer extends Component {
  menuItemClicked(value) {
    this.props.router.push(value)
    this.props.toggleNav()
  }

  render() {
    return (
      <Drawer
        open={this.props.open}
        docked={true}
        onRequestChange={this.props.toggleNav}
      >
        <div onClick={this.menuItemClicked.bind(this, '/')} style={this.props.styles.logo}>
          Course Manager
        </div>
        <MenuItem onClick={this.menuItemClicked.bind(this,'/users')}>Students</MenuItem>
        <MenuItem onClick={this.menuItemClicked.bind(this,'/workshops')}>Courses</MenuItem>
      </Drawer>
    )
  }
}

AppNavDrawer.propTypes = {
  toggleNav: React.PropTypes.func.isRequired,
  styles: React.PropTypes.object.isRequired,
  router: React.PropTypes.object.isRequired
}

export default withRouter(AppNavDrawer)
