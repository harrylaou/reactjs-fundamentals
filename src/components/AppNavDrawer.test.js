import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import AppNavDrawer from './AppNavDrawer'

import getMuiTheme from 'material-ui/styles/getMuiTheme'

describe('<AppNavDrawer />', () => {
  it('should render with default props', () => {
    const toggleNav = jest.fn();
    const push = jest.fn();
    const props = {
      styles: {},
      open: false,
      toggleNav
    }

    const wrapper = shallow(
      <AppNavDrawer {...props} />,
      {context: {
        muiTheme: getMuiTheme(),
        router: {push}
      }}
    )

    expect(shallowToJson(wrapper)).toMatchSnapshot()

    wrapper.find('MenuItem').at(0).simulate('click')

    expect(push).toBeCalledWith('/users')
    expect(toggleNav).toBeCalled()
  })
})