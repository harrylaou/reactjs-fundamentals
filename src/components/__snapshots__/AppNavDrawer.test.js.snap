// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppNavDrawer /> should render with default props 1`] = `
<Drawer
  disableSwipeToOpen={false}
  docked={true}
  onRequestChange={[Function]}
  open={false}
  openSecondary={false}
  swipeAreaWidth={30}
  width={null}
  zDepth={2}
>
  <div
    onClick={[Function]}
  >
    Course Manager
  </div>
  <MenuItem
    checked={false}
    desktop={false}
    disabled={false}
    focusState="none"
    insetChildren={false}
    onClick={[Function]}
  >
    Students
  </MenuItem>
  <MenuItem
    checked={false}
    desktop={false}
    disabled={false}
    focusState="none"
    insetChildren={false}
    onClick={[Function]}
  >
    Courses
  </MenuItem>
</Drawer>
`;
